unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -7275,6 +7223,160 @@ ProceduralAsset asset
+
+            // Get the agent's bounds
+            var bounds = GetAgentBounds(agent, agentType);
+
+            Debug.Log($"the global position of the agent bounds is: {bounds.center:F8}");
+
+            // Check if the spawned boxCollider is colliding with other objects
+            int layerMask = LayerMask.GetMask("SimObjVisible", "Procedural1", "Procedural2", "Procedural3", "Procedural0");
+            
+            Vector3 newBoxCenter = new Vector3(bounds.center.x - 100f, bounds.center.y - 100f, bounds.center.z - 100f);
+            Debug.Log($"the center for the new box should be at the agent's original position but is: {newBoxCenter:F8}");
+
+            #if UNITY_EDITOR
+            /////////////////////////////////////////////////
+            this.baseAgentComponent.boxCenter = newBoxCenter;
+            this.baseAgentComponent.boxHalfExtents = bounds.extents;
+            this.baseAgentComponent.boxOrientation = originalRotation;
+            this.baseAgentComponent.drawBox = true;
+
+            //for visualization lets spawna cube at the center of where the boxCenter supposedly is
+            GameObject cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cube.transform.position = this.baseAgentComponent.boxCenter;
+            cube.transform.rotation = this.baseAgentComponent.boxOrientation;
+            cube.transform.localScale = new Vector3(0.2f, 0.2f, 0.2f); //scale is small just so we can see it a little
+
+
+            Debug.Log("draw gizmos set for debug draw!");
+            ////////////////////////////////////////////////
+            #endif
+
+            if (Physics.CheckBox(newBoxCenter, bounds.extents, originalRotation, layerMask)) {
+                this.transform.position = originalPosition;
+                this.transform.rotation = originalRotation;
+                throw new InvalidOperationException(
+                    "Spawned box collider is colliding with other objects. Cannot spawn box collider."
+                );
+            }
+
+            // Move the agent to the pose aligned with bounds' center and rotation
+            // this.transform.position = new Vector3(originalPosition.x + 100f, originalPosition.y + 100f, originalPosition.z + 100f);
+            // this.transform.rotation = Quaternion.identity;
+
+            // Spawn the box collider
+            Vector3 colliderSize = new Vector3(
+                scaleRatio.x * bounds.size.x,
+                scaleRatio.y * bounds.size.y,
+                scaleRatio.z * bounds.size.z
+            );
+            if (useAbsoluteSize) {
+                colliderSize = scaleRatio;
+            }
+
+            GameObject noneTriggeredEncapsulatingBox = new GameObject("NonTriggeredEncapsulatingBox");
+            // noneTriggeredEncapsulatingBox.transform.position = new Vector3(bounds.center.x, bounds.center.y, agent.transform.position.z);
+            noneTriggeredEncapsulatingBox.transform.position = bounds.center;
```

Comment:
This is set to `bounds.center` which computed using agent position  with the added +100, wouldn't we want `newBoxCenter` here which removes the offset?

Same comment for below's `triggeredEncapsulatingBox` position.